### PR TITLE
Migrate mongoose plugin to opentelemetryjs-contrib

### PR DESCRIPTION
**Pull request by wdalmut**
As mentioned in #18 i propose the mongoose plugin migration to the opentelemetryjs-contrib repository.

Currently this plugin is available in the opentelemetry registry: https://opentelemetry.io/registry/instrumentation-js-mongoose/

### CODE REVIEW

**Comment by wdalmut** [comment]
i have to fix all issues i just need time to close everything
/cc @vmarchaud

**Comment by vmarchaud** [comment]
@wdalmut No worries, take the time you need. If you need any help tell me

File: plugins/node/opentelemetry-plugin-mongoose/.travis.yml
Comment (wdalmut) [inline plugins/node/opentelemetry-plugin-mongoose/.travis.yml:1]: i have to drop this...

File: plugins/node/opentelemetry-plugin-mongoose/LICENSE
Comment (wdalmut) [inline plugins/node/opentelemetry-plugin-mongoose/LICENSE:1]: Need to update this file

**Comment by wdalmut** [comment]
@vmarchaud i think this could be my first proposal...

File: plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts
Comment (vmarchaud) [inline plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts:53]: i think you need to update the reference to the issue

File: plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts
Comment (vmarchaud) [inline plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts:239]: we generally call this reference `plugin`

File: plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts
Comment (vmarchaud) [inline plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts:209]: what is `product` here ?

File: plugins/node/opentelemetry-plugin-mongoose/jasmine.json
Comment (vmarchaud) [inline plugins/node/opentelemetry-plugin-mongoose/jasmine.json:1]: we currently use mocha for all of our tests so i think those tests would need to be converted (not sure if its a hard requirement though)

File: plugins/node/opentelemetry-plugin-mongoose/package.json
Comment (vmarchaud) [inline plugins/node/opentelemetry-plugin-mongoose/package.json:41]: could you bump tose versions to latest ?

File: plugins/node/opentelemetry-plugin-mongoose/README.md
Comment (vmarchaud) [inline plugins/node/opentelemetry-plugin-mongoose/README.md:1]: This need to match the style of ours readmes, example with mongodb: https://github.com/open-telemetry/opentelemetry-js-contrib/blob/master/plugins/node/opentelemetry-plugin-mongodb/README.md

File: plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts
Comment (vmarchaud) [inline plugins/node/opentelemetry-plugin-mongoose/src/mongoose.ts:54]: Currently exporting here means end-user could have access to the symbol, i think you should move it inside the utils and importing it here

**Comment by wdalmut** [comment]
I have fixed all notes and moved the whole test suite to mocha

//cc @vmarchaud

**Comment by wdalmut** [comment]
> I have fixed all notes and moved the whole test suite to mocha
> 
> //cc @vmarchaud

i miss some comment in the review but i think i can solve everything today...

**Review by wdalmut — COMMENTED** [review]
All comments now are included

**Review by vmarchaud — APPROVED** [review]

**Comment by CyberHippo** [comment]
@wdalmut Thank you for your great work! Could you please merge it?

**Comment by wdalmut** [comment]
@vmarchaud do you have updates on this?

**Comment by vmarchaud** [comment]
@wdalmut Well i was fine with merging this at the time, but it seems we have a problem for reviewing PR on this repo. In the mean time we updated the way plugin are implemented to fix few issues (see [the new base package](https://github.com/open-telemetry/opentelemetry-js/tree/master/packages/opentelemetry-instrumentation)).

However the new system is not yet finished (we dont have a loader for those) so i think it would be better to wait, update the plugin to use the new system and then merge it on the contrib repo.

**Comment by obecny** [comment]
@dyladan @vmarchaud what should be done with this plugin as it is outdated anyway ?
1. update to latest, but don't convert yet to instrumentation, once it is done, then check it and finally merge, then convert it to instrumentation
2. refactor this to be an instrumentation, update to latest, review, merge,
3. any other options ?

**Comment by vmarchaud** [comment]
I woud prefer to not merge it so we can get rid of all plugins to release 1.0 of SDK. So i'll be in favor of closing this one and create an issue to port it to instrumentation.

**Comment by vmarchaud** [comment]
I'm going to close this as we are deprecating the `BasePlugin` system, if people are looking for an instrumentation you should checkout https://github.com/aspecto-io/opentelemetry-ext-js/tree/master/packages/instrumentation-mongoose

**Pull request closed without merging**
